Supportzipfixes

bendk opened the pull request.

Commit:
bz17898: don't include cover art in crash report zip files

Commit:
bz18835: force zipfile filenames to be ASCII.

Fixed code handling when the path is a non-ascii bytestring like utf-8.  Added
a comment on why we need it.

Commit:
Refactored code to create support directory backups.

I figured it's getting complicated enough that it can use its own class.
Also, this lets us add unittests.

Commit:
bz18829: limit the size of zip files we upload te bogon deflector.

paulswartz commented on tv/lib/test/utiltest.py:
```diff
@@ -908,3 +910,90 @@ def test_subdirs(self):
+                                       max_size=1000000)
+        archive = zipfile.ZipFile(backup.fileobj(), 'r')
+        errors = archive.testzip()
+        if errors is not None:
```
The regular way to write this (in Python < 2.7, at least) is `self.assertTrue(errors is None, msg)`.

paulswartz commented on tv/lib/test/utiltest.py:
```diff
@@ -908,3 +910,90 @@ def test_subdirs(self):
+        backup = util.SupportDirBackup(self.support_dir, self.skip_dirs,
+                                       max_size=max_size)
+        filesize = os.stat(backup.backupfile).st_size
+        if filesize > 1100000:
```
Same with this: `self.assertTrue(filesize < 1100000, msg)`.

paulswartz commented on tv/lib/util.py:
```diff
@@ -1162,3 +1164,77 @@ def split_values_for_sqlite(value_list):
+    def __init__(self, support_dir, skip_dirs, max_size):
+        logging.info("Attempting to back up support directory: %r",
+                     support_dir)
+        uniqfn = "%012ddatabasebackup.zip" % random.randrange(0, 999999999999)
```
Maybe do this with `time.time()` or something, so that they're ordered?

bendk replied:
I think I'll change it to next_free_filename() so they're ordered and  it matches how we generate other filenames like this.

paulswartz commented on tv/lib/util.py:
```diff
@@ -1162,3 +1164,77 @@ def split_values_for_sqlite(value_list):
+        logging.info("Attempting to back up support directory: %r",
+                     support_dir)
+        uniqfn = "%012ddatabasebackup.zip" % random.randrange(0, 999999999999)
+        self.backupfile = os.path.join(tempfile.gettempdir(), uniqfn)
```
Should we check that the file doesn't exist already?

paulswartz commented on tv/lib/util.py:
```diff
@@ -1162,3 +1164,77 @@ def split_values_for_sqlite(value_list):
+
+        total_size = 0
+        for root, dummy, files in os.walk(support_dir):
+            if self.should_skip_directory(root):
```
This works, but it might be more efficient if you filtered the unneeded directories from `dirnames` (called `dummy` here).

``` python
remove = [f for f in dummy if self.should_skip_directory(f)]
for f in remove:
    dummy.remove(f)
```

You'd also be able to remove the path normalizing stuff in `should_skip_directory`.

paulswartz commented on tv/lib/util.py:
```diff
@@ -1162,3 +1164,77 @@ def split_values_for_sqlite(value_list):
+                     self.backupfile, os.path.getsize(self.backupfile))
+
+    def should_skip_directory(self, directory):
+        if os.path.islink(directory):
```
`os.walk()` doesn't recurse into links by default, so this isn't needed.

Commit:
Cleanups based on pauls review.

Started using assertTrue in unittests.

Use next_free_filename() to generate unique filenames.  This ensures that the
filename is really unused and also makes it so that the paths are ordered.
Moved next_free_filename from download_utils to util.

Fixes for walking through directories.
  - Don't include a check for symlinks.  os.walk() does that for us.
  - Don't send the paths we get from os.walk() through normpath.  They should
    be normalized already.
  - Removed directories from the list that os.walk() passes us.  This makes it
    so that os.walk() doesn't recurse into these at all and is more efficient
    than skipping the directory after os.walk() walks into it.

Commit:
Adding back in the check if a file is a link.

paulswartz commented:
Looks good.
